backend/src/route.rs

```diff
@@ -177,3 +198,106 @@ impl Route {
         (distance, time)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::osm_tests::load_osm_xml;
+
+    fn r(road_id: usize) -> RoadID {
+        RoadID(road_id)
+    }
+
+    #[test]
+    fn basic_route() {
+        // Helper to set up testing scenario. It's really verbose. We should probably do something else.
```

Comment:
This comment can be removed, since the osm.xml tests are used now?

I like the ascii diagrams!